src/material-experimental/mdc-list/list-styler-elements.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, HostBinding, Inject, InjectionToken, Optional} from '@angular/core';
+
+/**
+ * Interface describing a list option. This is used to avoid circular
+ * dependencies between the list-option and the styler directives.
+ * @docs-private
+ */
+export interface ListOption {
+  _getCheckboxPosition(): 'before'|'after';
+}
+
+/**
+ * Injection token that can be used to reference instances of an `ListOption`. It serves
+ * as alternative token to an actual implementation which could result in undesired
+ * retention of the class or circular references breaking runtime execution.
+ * @docs-private
+ */
+export const LIST_OPTION = new InjectionToken<ListOption>('ListOption');
+
+/**
+ * Directive whose purpose is to add the MDC list-item `graphic` or `meta` class
+ * depending on checkbox position in list options.
+ */
+@Directive()
+export abstract class _MatListItemGraphicAlignmentBase {
+  @HostBinding('class.mdc-list-item__graphic')
+  get alignAtStart(): boolean {
```

Comment:
Instead of having the two other directies extending this class, couldn't you change the selector so it matches the avatar and icon? That way you can get rid of the `extends`, the getters, and place the binding in `host`.

Reply:
This will require us to duplicate the directive selectors. I think both ways have trade-offs here, but I went with what you proposed since I'd like to avoid the getters.